share/spice/sales_tax/sales_tax.js

```diff
@@ -1,77 +1,56 @@
-            
-            if(apiObjectName == statNameTmp) {
-               taxRate = api_result.rates[i].rate;
-            // workaround washington d.c = district of columbia
-            } else if(apiObjectName == "district of columbia" && statNameTmp == "washington d.c") {
-                taxRate = api_result.rates[i].rate;
-            }
-
+        if (!api_result || api_result.header.status !== "SUCCESS") {
+            return Spice.failed('sales_tax2');
         }
 
-        // Fail if we have no rate (edge case)
-        if (taxRate === undefined) {
-            return Spice.failed('sales_tax');
-        }
-
-        // title and subtitle
-        titleResult = taxRate+"%";
-        subtitleResult = stateName+" - "+"State Tax"
-
-        // Fix for sourceUrl washington d.c => washington-dc
-        // clean stateName for use as sourceUrl
-        stateName = stateName.replace(/\./,"").replace(/\s/,"-").toLowerCase();
-
         // Display
         Spice.add({
             id: "sales_tax",
-            name: "Answer",
-            data: titleResult,
+            name: "Finance",
+            data: api_result.stateSalesTaxInfos,
             meta: {
-                sourceName: "Avalara",
-                sourceUrl: 'http://www.taxrates.com/state-rates/' + stateName
+                //itemType: "Results",
+                //searchTerm: api_result.query,
+                sourceName: "snapCX.io",
+                sourceIcon: false,
+                sourceIconUrl: "http://snapcx.io/favicon.ico",
+                sourceUrl: "https://snapcx.io"
             },
             normalize: function(item) {
+                var stateName, titleResult, subtitleResult, minTaxRate, maxTaxRate, noTaxState;
+
+                stateName   = item.stateName;
+                minTaxRate  = (item.minimumTaxRate * 100).toFixed(2) +"%";
+                maxTaxRate  = (item.maximumTaxRate * 100).toFixed(2) +"%";
+                noTaxState  = item.noTaxState;
+                titleResult = stateName+" - "+"Sales Tax";
+                console.log("Title is "+titleResult);
+
+                if (noTaxState === false) {
+                    if (minTaxRate === maxTaxRate) {
+                        subtitleResult = "Sales Tax is "+minTaxRate;
+                    } else {
+                        subtitleResult = "Sale Tax ranges from "+minTaxRate+" to "+maxTaxRate;
+                    }
+                } else {
+                    subtitleResult = "No sales tax for this state";
+                }
                 return {
                     title: titleResult,
-                    subtitle: subtitleResult
+                    description: subtitleResult                    
                 };
             },
             templates: {
                 group: 'text',
                 options: {
-                    moreAt: true
-                }
+                    moreAt: true                    
+                },
+                detail: false,
```

Comment:
We should render a detail pane for single results. I suspect the majority of the queries for this IA will specify the State so a single tile isn't ideal. We should try and maintain/mimic the current design.